### PR TITLE
feat(friendshipper): add support for multiple playtest regions

### PR DESCRIPTION
**Pull request by rudoi**
This enables switching between regions for deploying playtests + servers. Enabled regions are enumerated as part of dynamic config. 

At present the default is still `us-west-2`.

### CODE REVIEW

File: friendshipper/src/lib/components/preferences/PreferencesModal.svelte [hunk 1]
```diff
@@ -226,6 +235,20 @@
 				For engineers. Enable if you want to debug the game client locally. Increases download size.
 			</Tooltip>
 		</div>
+		{#if $dynamicConfig.playtestRegions.length > 1}
```
Comment (sylviacx) [inline friendshipper/src/lib/components/preferences/PreferencesModal.svelte:238]: If we only had one playtest region, should we still show it in preferences so users know which region they're in?

File: friendshipper/src/routes/servers/+page.svelte [hunk 2]
```diff
@@ -42,6 +42,10 @@
 		}
 	};
 
+	void listen('preferences-closed', () => {
```
Comment (sylviacx) [inline friendshipper/src/routes/servers/+page.svelte:45]: What does this fn do, how is it used if it's not called directly on this page?

Comment (rudoi) [reply friendshipper/src/routes/servers/+page.svelte:45]: The preferences modal itself emits a `preferences-closed` event, which this component listens to.

File: friendshipper/src/lib/components/preferences/PreferencesModal.svelte (hunk 1 above)
Comment (rudoi) [reply friendshipper/src/lib/components/preferences/PreferencesModal.svelte:238]: I think I could go either way. In my head if there's only one region, I don't think the info is useful for typical users.

Comment (sylviacx) [reply friendshipper/src/lib/components/preferences/PreferencesModal.svelte:238]: gotcha, we can keep as is then

**Review by sylviacx — APPROVED** [review]

**Pull request merged**